src/main/java/eu/europeana/metis/sandbox/service/debias/DebiasMachineService.java

```diff
@@ -0,0 +1,80 @@
+package eu.europeana.metis.sandbox.service.debias;
+
+import eu.europeana.metis.sandbox.dto.debias.DetectionInfoDto;
+import eu.europeana.metis.sandbox.entity.debias.DetectionEntity;
+import eu.europeana.metis.sandbox.repository.debias.DetectRepository;
+
+public class DebiasMachineService implements DetectService {
+
+  private final Stateful ready;
+  private final Stateful processing;
+  private final Stateful completed;
+  private final Stateful error;
+  private final DetectRepository detectRepository;
+  private Stateful state;
+
+  public DebiasMachineService(DetectRepository detectRepository) {
+    this.ready = new ReadyState(this, detectRepository);
+    this.processing = new ProcessingState(this, detectRepository);
+    this.completed = new CompletedState(this, detectRepository);
+    this.error = new ErrorState(this, detectRepository);
+    this.state = this.ready;
+    this.detectRepository = detectRepository;
+  }
+
+  @Override
+  public void fail(Long datasetId) {
+    state.fail(datasetId);
+  }
+
+  @Override
+  public void success(Long datasetId) {
+    state.success(datasetId);
+  }
+
+  @Override
+  public boolean process(Long datasetId) {
```

Comment:
This time I decided not to go "simple" in a sense because what I have learned from you. In the end I see it as simple, I took the time to stop and think as we had in our conversations (I don´t want to have something just for the rush that in the end will take more time fixing).  So, the implementation follows the design of a state machine. 
![Debias-Transitions-State](https://github.com/user-attachments/assets/c83af209-1706-44e2-a0fb-9c98d9279558)

This design creates the structure to allow in every state to perform an operation.
**Ready**, here the intention is to gather the records/information, that will be sent to the Debias API and do the proper validations in here .
**Processing**, here the intention is to process the gathered information.
**Error**, here is to keep record why the processing failed and gather information.
**Completed**, here the intention is to do the clean up or finishing processes.

This gives room for small units of work in each class. 

with the design I can associate a description for each state.
**Ready**, "DeBias detection not started".
**Processing**, "DeBias detection in progress".
**Error", "DeBias detection failed".
**Completed**, "DeBias detection completed".

The asynchronous call I can use this on the next ticket of the implementation.

Reply:
Good that you pointed out this flow diagram, I was wondering where the "state machine" came from. :D
But I disagree with the simplicity.
The flow of states does not have to be that complex in code as long as the "flow" is accurate and it can be with code you already have, right?

Reply:
Correct. They don't have to be complex and that is not my intention. I'm looking from the perspective that each class will end up to be simple.